src/library/searchqueryparser.h

```diff
@@ -11,14 +11,19 @@
 
 class SearchQueryParser {
   public:
-    SearchQueryParser(QSqlDatabase& database);
+    SearchQueryParser(TrackCollection* pTrackCollection);
```

Comment:
Single argument constructors should be declared 'explicit by default! Otherwise the compiler will apply an implicit conversion from TrackCollection* to SearchQuery whenever it needs to without any warning.